src/components/GraphiQL/GraphiQL.tsx

```diff
@@ -0,0 +1,346 @@
+  // @ts-ignore
+  [Key in keyof Obj as `${string & Key}${Suffix}`]: Obj[Key];
+};
+
+export type GraphiQLInterfaceProps = WriteableEditorProps &
+  AddSuffix<Pick<UseQueryEditorArgs, "onEdit">, "Query"> &
+  Pick<UseQueryEditorArgs, "onCopyQuery"> &
+  AddSuffix<Pick<UseVariableEditorArgs, "onEdit">, "Variables"> &
+  AddSuffix<Pick<UseHeaderEditorArgs, "onEdit">, "Headers"> &
+  Pick<UseResponseEditorArgs, "responseTooltip"> & {
+    children?: ReactNode;
+    defaultEditorToolsVisibility?: boolean | "variables" | "headers";
+    isHeadersEditorEnabled?: boolean;
+    toolbar?: GraphiQLToolbarConfig;
+  };
+
+export function GraphiQLInterface(props: GraphiQLInterfaceProps) {
+  const editorContext = useEditorContext({ nonNull: true });
+  const pluginContext = usePluginContext();
+
+  const classes = useStyles({});
+
+  const copy = useCopyQuery({ onCopyQuery: props.onCopyQuery });
+  const prettify = usePrettifyEditors();
+
+  const PluginContent = pluginContext?.visiblePlugin?.content;
+
+  const pluginResize = useDragResize({
+    defaultSizeRelation: 1 / 3,
+    direction: "horizontal",
+    initiallyHidden: pluginContext?.visiblePlugin ? undefined : "first",
+    sizeThresholdSecond: 200,
+    storageKey: "docExplorerFlex",
+  });
+  const editorResize = useDragResize({
+    direction: "horizontal",
+    storageKey: "editorFlex",
+  });
+  const editorToolsResize = useDragResize({
+    defaultSizeRelation: 3,
+    direction: "vertical",
+    sizeThresholdSecond: 60,
+    storageKey: "secondaryEditorFlex",
+  });
+
+  const children = React.Children.toArray(props.children);
+
+  const toolbar = children.find(child =>
+    isChildComponentType(child, GraphiQL.Toolbar),
+  ) || (
+    <>
+      <ToolbarButton
+        onClick={() => prettify()}
+        label="Prettify query (Shift-Ctrl-P)"
+      >
+        <PrettifyIcon className="graphiql-toolbar-icon" aria-hidden="true" />
+      </ToolbarButton>
+      <ToolbarButton onClick={() => copy()} label="Copy query (Shift-Ctrl-C)">
+        <CopyIcon className="graphiql-toolbar-icon" aria-hidden="true" />
+      </ToolbarButton>
+      {props.toolbar?.additionalContent || null}
+    </>
+  );
+
+  const onClickReference = () => {
+    if (pluginResize.hiddenElement === "first") {
+      pluginResize.setHiddenElement(null);
+    }
+  };
+
+  const theme = useTheme();
+  const rootStyle = {
+    "--font-size-body": theme.typography.body2.fontSize,
+    "--font-size-h2": theme.typography.h3.fontSize,
+    "--font-size-h3": theme.typography.h3.fontSize,
+    "--font-size-h4": theme.typography.h4.fontSize,
+    "--font-size-hint": theme.typography.caption.fontSize,
+    "--font-size-inline-code": theme.typography.caption.fontSize,
+  } as React.CSSProperties;
+
+  const { theme: graphiqlTheme, setTheme: setGraphiqlTheme } =
+    useGraphiqlTheme();
+  useEffect(() => {
+    if (theme.themeType !== graphiqlTheme) {
+      setGraphiqlTheme(theme.themeType);
+    }
+  });
```

Comment:
similar here, looks like theming context

**suggestion**: separated hook, and re-use it

Reply:
in the previous one you have purely editor styling/resizing, this one is more like theme itself (just emphasising that it's a separated responsibility)